decoder_test.go

```diff
@@ -400,3 +400,34 @@ func TestCompactIntegersBoundaries(t *testing.T) {
 		}
 	})
 }
+
+func decodeTest[T any](t *testing.T, value []byte, expect T) {
+	buf := bytes.NewBuffer(value)
+	dec := NewDecoder(buf)
+	switch typed := any(expect).(type) {
+	case uint8:
```

Comment:
It's a weird mix, the function is generic but uses dynamic dispatch to find the type. How about taking `any` instead?
```suggestion
func decodeTest(t *testing.T, value []byte, expect any) {
	buf := bytes.NewBuffer(value)
	dec := NewDecoder(buf)
	switch typed := expect.(type) {
	case uint8:
```

Btw, IMHO this should just be inlined in the test body.

Reply:
Simplified the test (see test-related comment below)